tcl/generator-htmlbrowser/htmlbrowser_template.tcl

```diff
@@ -0,0 +1,430 @@
+            return true
+        } else {
+            set current [$current parent]
+        }
+    }
+    return false
+}
+
+proc indent count {
+    set result {}
+    while {$count > 0} {
+        set result "$result    "
+        set count [expr {$count - 1}]
+    }
+    return $result
+}
+
+proc generateNavigationRec {current activeItem count} {
+    ## if $current is a group or registerfile
+    if {[$current isa osys::rfg::Group]} {
+        if {!([string compare $current $activeItem])} {
+            puts "[indent $count]<a href=\"#\" class=\"list-group-item active\">"
+        } else {
+            puts "[indent $count]<a href=\"[getFileName $current $activeItem]\" class=\"list-group-item\">"
+        }
+        set count [expr {$count + 1}]
+        if {[subCompIsActive $current $activeItem]} {
+            puts "[indent $count]<span class=\"glyphicon glyphicon-chevron-down clickable\" data-toggle=\"collapse\" href=\"#[getAbsoluteName $current _]\"></span>[$current name]"
+        } else {
+            puts "[indent $count]<span class=\"glyphicon glyphicon-chevron-right clickable\" data-toggle=\"collapse\" href=\"#[getAbsoluteName $current _]\"></span>[$current name]"
+        }
+        puts "[indent $count]<span class=\"badge\">0x[format %x [$current getAttributeValue software.osys::rfg::absolute_address]]</span>"
+        set count [expr {$count - 1}]
+        puts "[indent $count]</a>"
+        if {[subCompIsActive $current $activeItem]} {
+            puts "[indent $count]<div class=\"list-group collapse in\" id=\"[getAbsoluteName $current _]\">"
+        } else {
+            puts "[indent $count]<div class=\"list-group collapse\" id=\"[getAbsoluteName $current _]\">"
+        }
+        $current onEachComponent {
+            generateNavigationRec $it $activeItem [expr {$count + 1}]
+        }
+        puts "[indent $count]</div>"
+    } else {
+        ## if $current is the active component
+        if {!([string compare $current $activeItem])} {
+            puts "[indent $count]<a href=\"#\" class=\"list-group-item active\">[$current name]"
+        } else {
+            puts "[indent $count]<a href=\"[getFileName $current $activeItem]\" class=\"list-group-item\">[$current name]"
+        }
+        set count [expr {$count + 1}]
+        puts "[indent $count]<span class=\"badge\">0x[format %x [$current getAttributeValue software.osys::rfg::absolute_address]]</span>"
+        set count [expr {$count - 1}]
+        puts "[indent $count]</a>"
+    }
+}
+
+proc getType item {
+    if {[$item isa osys::rfg::Register]} {
+        return "Register"
+    } elseif {[$item isa osys::rfg::RamBlock]} {
+        return "RAM Block"
+    } elseif {[$item isa osys::rfg::Group]} {
+        if {[$item isa osys::rfg::RegisterFile]} {
+            return "Register File"
+        } else {
+            return "Group"
+        }
+    } else {
+        return "-"
+    }
+}
+
+proc generateSwAttr item {
```

Comment:
There should be a function to get a list of all attributes (attr_list) of an item withing an attribute group. This should be used to make the html generator more maintainable. 

If then a new attribute is added the html generator does not have to be modified.